contracts/src/libraries/LPMath.sol

```diff
@@ -76,14 +134,21 @@ library LPMath {
         // zeta_new = zeta_old * (z_new / z_old)
         if (_shareAdjustment >= 0) {
             // NOTE: Rounding down to have a larger effective share reserves.
-            shareAdjustment = shareReserves
+            (shareAdjustment, success) = shareReserves
                 .mulDivDown(uint256(_shareAdjustment), _shareReserves)
-                .toInt256();
+                .toInt256Safe();
+            if (!success) {
+                return (0, 0, 0, false);
+            }
         } else {
             // NOTE: Rounding up to have a larger effective share reserves.
-            shareAdjustment = -shareReserves
+            (shareAdjustment, success) = shareReserves
                 .mulDivUp(uint256(-_shareAdjustment), _shareReserves)
-                .toInt256();
+                .toInt256Safe();
+            if (!success) {
+                return (0, 0, 0, false);
+            }
+            shareAdjustment = -shareAdjustment;
```

Comment:
this is not not confusing ;)

Reply:
Yeah I get it. It made more sense before when it was all on one line. I'll see if there is a better way to lay this out.

Reply:
I added a comment for this.